nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/merge/RecordBinManager.java

```diff
@@ -195,6 +197,9 @@ private RecordBinThresholds createThresholds() {
         final String mergeStrategy = context.getProperty(MergeRecord.MERGE_STRATEGY).getValue();
         if (MergeRecord.MERGE_STRATEGY_DEFRAGMENT.getValue().equals(mergeStrategy)) {
             fragmentCountAttribute = MergeContent.FRAGMENT_COUNT_ATTRIBUTE;
+            if (!StringUtils.isEmpty(fragmentCountAttribute)) {
```

Comment:
This if statement always return true as it checks `MergeContent. FRAGMENT_COUNT_ATTRIBUTE` constant value. I assume you wanted to check whether `flowfile.getAttribute(fragmentCountAttribute)` is not null/empty?

Reply:
Oh man, I'm glad you caught that!! Thank you, you are exactly right. Checking in a fix now!

Reply:
Oh man, I'm glad you caught that!! Thank you, you are exactly right. Checking in a fix now!

Reply:
You are exactly right!! Thank you for catching that!

Fix is checked in, but as per the contributor's guide I squashed it so it just shows up as a single checkin.